packages/block-library/src/navigation/placeholder.js

```diff
@@ -56,6 +56,12 @@ function mapMenuItemsToBlocks( menuItems ) {
 				? __( '(no title)' )
 				: menuItem.title.rendered,
 			opensInNewTab: menuItem.target === '_blank',
+			type: menuItem?.object,
+			...( menuItem?.object !== 'custom' && {
+				id: menuItem?.object_id,
+			} ),
+			// the equivalent on `nav_menu_item` is "type" but it is stored with underscores whereas the block attr uses hyphens.
+			kind: menuItem.type.replace( '_', '-' ),
```

Comment:
The `kind` attribute on the block basically maps to [the `type` attribute on the nav menu item](https://core.trac.wordpress.org/browser/tags/5.7.1/src/wp-includes/nav-menu.php#L802). The main difference is that the values use `_` instead of `-` as separators.

Note that this is _not_ an identifier of the object type (ie: `Post`, `Page`, `Category`, `{Custom Post Type} Category`...etc) but rather an enum of:

* [`post-type`](https://core.trac.wordpress.org/browser/tags/5.7.1/src/wp-includes/nav-menu.php#L822).
* [`post-type-archive`](https://core.trac.wordpress.org/browser/tags/5.7.1/src/wp-includes/nav-menu.php#L862)
* [`taxonomy`](https://core.trac.wordpress.org/browser/tags/5.7.1/src/wp-includes/nav-menu.php#L877)

The use of `custom` is simply a convention rather than part of the `enum` which allows users to set a custom link without having to associate it with any given "entity" within WordPress. 

I would have thought it would be best for the block to use `custom` as the fallback case.

> Note that while the custom link variation is an option in the block editor, this really means that we don't set initial search parameters

This makes sense because you aren't searching for an entity.

> ...or a specific type like a post link.

I don't follow this sorry. Can you help me to understand?